libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -238,7 +238,10 @@ class ObjectBuffer {
     // !!! Note ObjectBuffer_TS is a duplicate of this, update in both places !!!
     const T *readptr(uint32_t &n) {
         uint32_t avail_bytes = 0;
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wcast-align"
```

Comment:
you should do the same L414 as said in the comment

Reply:
ha! nice catch